testing/smoke/supported-os/main.tf

```diff
@@ -0,0 +1,218 @@
+    "RHEL-8"                           = "309956199498" # Red Hat
+  }
+  image_ssh_users = {
+    "ubuntu-bionic-18.04-amd64-server" = "ubuntu"
+    "ubuntu-focal-20.04-amd64-server"  = "ubuntu"
+    "ubuntu-jammy-22.04-amd64-server"  = "ubuntu"
+    "debian-10-amd64"                  = "admin"
+    "debian-11-amd64"                  = "admin"
+    "amzn2-ami-kernel-5.10"            = "ec2-user"
+    "RHEL-7"                           = "ec2-user"
+    "RHEL-8"                           = "ec2-user"
+  }
+  apm_port = "8200"
+}
+
+data "aws_ami" "os" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["*${var.aws_os}*"]
+  }
+
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+
+  filter {
+    name   = "root-device-type"
+    values = ["ebs"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  owners = [local.image_owners[var.aws_os]]
+}
+
+resource "aws_security_group" "main" {
+  egress = [
+    {
+      cidr_blocks      = ["0.0.0.0/0", ]
+      description      = ""
+      from_port        = 0
+      ipv6_cidr_blocks = []
+      prefix_list_ids  = []
+      protocol         = "-1"
+      security_groups  = []
+      self             = false
+      to_port          = 0
+    }
+  ]
+  ingress = [
+    {
+      cidr_blocks      = ["0.0.0.0/0", ]
+      description      = ""
+      from_port        = 22
+      ipv6_cidr_blocks = []
+      prefix_list_ids  = []
+      protocol         = "tcp"
+      security_groups  = []
+      self             = false
+      to_port          = 22
+    },
+    {
+      cidr_blocks      = ["0.0.0.0/0", ]
+      description      = ""
+      from_port        = local.apm_port
+      ipv6_cidr_blocks = []
+      prefix_list_ids  = []
+      protocol         = "tcp"
+      security_groups  = []
+      self             = false
+      to_port          = local.apm_port
+    }
+  ]
+}
+
+resource "aws_instance" "apm" {
+  ami           = data.aws_ami.os.id
+  instance_type = "t3.micro"
+  key_name      = aws_key_pair.provisioner_key.key_name
+
+  connection {
+    type        = "ssh"
+    user        = local.image_ssh_users[var.aws_os]
+    host        = self.public_ip
+    private_key = file("${var.aws_provisioner_key_name}")
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+      "curl ${data.external.getlatestapmserver.result.deb} -o apm-server.deb && curl ${data.external.getlatestapmserver.result.rpm} -o apm-server.rpm",
+      "sudo dpkg -i apm-server.deb || sudo yum -y install apm-server.rpm",
+      "sudo rm /etc/apm-server/apm-server.yml",
+      "echo \"apm-server:\n  host: \\\"0.0.0.0:${local.apm_port}\\\"\noutput:\n  elasticsearch:\n    hosts: [\\\"${module.ec_deployment.elasticsearch_url}\\\"]\n    username: \\\"${module.ec_deployment.elasticsearch_username}\\\"\n    password: \\\"${module.ec_deployment.elasticsearch_password}\\\"\" | sudo tee -a /etc/apm-server/apm-server.yml",
```

Comment:
Should we be injecting the secret token here too?

Reply:
We don't need to. The APM server is not using the secret token afaik but the output variable is needed because the bash script sends the token in a http header.

Reply:
OK. I don't think it hurts either way, we can leave it for now.